ExtraDry/ExtraDry.Server/RuleEngine.cs

```diff
@@ -418,17 +527,66 @@ private static bool AttemptSoftDelete<T>(T item)
         }
         var property = type.GetProperty(softDelete.PropertyName, BindingFlags.Instance | BindingFlags.Public);
         if(property == null) {
-            throw new DryException($"Can't complete soft-delete, property '{softDelete.PropertyName}' does not exist as a public instance property of class '{type.Name}'.", "Can't delete item, internal application issue.");
+            return false;
         }
         try {
             property.SetValue(item, softDelete.DeleteValue);
         }
         catch {
-            throw new DryException($"Can't complete soft-delete, value provided is not convertable to type of property '{softDelete.PropertyName}", "Can't delete item, internal application issue.");
+            return false;
         }
         return true;
     }
 
-    private readonly IServiceProvider services;
+    private static bool AttemptSoftDelete(object[] items)
+    {
+        if(items == null) {
```

Comment:
Nullable enable is on, this should never fire...

Reply:
Removed